debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeOperationsTestIT.java

```diff
@@ -0,0 +1,220 @@
+        final Struct transformedUpdateValue = (Struct) transformedUpdate.value();
+
+        assertThat(transformedUpdate.valueSchema().field("id").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdate.valueSchema().field("dataInt").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdate.valueSchema().field("nested.dataInt").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdateValue.get("id")).isEqualTo(1);
+        assertThat(transformedUpdateValue.get("dataInt")).isEqualTo(246);
+        assertThat(transformedUpdateValue.get("nested.dataInt")).isEqualTo(100);
+    }
+
+    /**
+     * @see <a href="https://docs.mongodb.com/v3.6/reference/operator/update/mul/#up._S_mul">MongoDB operator update $mul</a>
+     */
+    @Test
+    public void shouldTransformOperationMul() throws InterruptedException {
+        SourceRecord updateRecord = executeSimpleUpdateOperation(
+                "{'$mul': {'dataInt': 3, 'nested.dataInt': 2, 'nonExistentField': 123}}"
+        );
+
+        final SourceRecord transformedUpdate = transformation.apply(updateRecord);
+        final Struct transformedUpdateValue = (Struct) transformedUpdate.value();
+
+        assertThat(transformedUpdate.valueSchema().field("id").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdate.valueSchema().field("dataInt").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdate.valueSchema().field("nested.dataInt").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdate.valueSchema().field("nonExistentField").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
+        assertThat(transformedUpdateValue.get("id")).isEqualTo(1);
+        assertThat(transformedUpdateValue.get("dataInt")).isEqualTo(369);
+        assertThat(transformedUpdateValue.get("nested.dataInt")).isEqualTo(246);
+        assertThat(transformedUpdateValue.get("nonExistentField")).isEqualTo(0);
+    }
+
+    /**
+     * @see <a href="https://docs.mongodb.com/v3.6/reference/operator/update/rename/#up._S_rename">MongoDB operator update $rename</a>
+     * Since the $rename operator generates internally $set and $unset operations the asserts won't pass
+     */
+    @Test
+    @Ignore("Depends on https://issues.jboss.org/browse/DBZ-612")
+    public void shouldTransformOperationRename() throws InterruptedException {
+        SourceRecord updateRecord = executeSimpleUpdateOperation(
+                "{'$rename': {'dataInt': 'dataIntNewName', 'nonExistentField': 'nonExistentFieldRenamed'}}"
+        );
+
+        final SourceRecord transformedUpdate = transformation.apply(updateRecord);
+        final Struct transformedUpdateValue = (Struct) transformedUpdate.value();
+
+        assertThat(transformedUpdate.valueSchema().field("id").schema()).isEqualTo(Schema.OPTIONAL_INT32_SCHEMA);
```

Comment:
How about fully assembling the expected value schema and then asserting on that (see `VerifyRecord#assertConnectSchemasAreEqual()`)?

Reply:
It's great! Should I just use or move to a more "generic place"? I'm afraid of crossing packages boundaries here